metricflow-semantics/metricflow_semantics/model/semantics/linkable_element.py

```diff
@@ -267,7 +265,8 @@ class SemanticModelJoinPath:
     would be represented by 2 path elements [(semantic_model0, A), (dimension_source1, B)]
     """
 
-    path_elements: Tuple[SemanticModelJoinPathElement, ...]
+    left_semantic_model_reference: SemanticModelReference
```

Comment:
With this change, `LinkableMetric.join_path.semantic_model_join_path` should no longer be an optional field (otherwise we won't know the `left_semantic_model_reference`). In the cases where it was previously None, it should now be a `SemanticModelJoinPath` that includes a `left_semantic_model_reference` and no path elements.